### PR TITLE
NIP-36 - sensitive content / content-warning

### PR DESCRIPTION
**Pull request by fernandolguevara**
[FR] Sensitive Content #64

### CODE REVIEW

**Comment by eskema** [comment]
this should be it's own NIP, doesn't belong in NIP1

File: 17.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+NIP-17
+======
+
+Sensitive Content / Content Warning
+-----------------------------------
+
+`draft` `optional` `author:fernandolguevara`
+
+Like in mastodon, `content-warning` tag enables users to specify if the event's content needs to be approved by readers to be shown.
```
Comment (eskema) [inline 17.md:9]: **Like in mastodon,** should be removed, it doesn't add anything...
The `content-warning` ...

File: 17.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Clients can show a blurred section in the content place.
```
Comment (eskema) [inline 17.md:10]: "Clients can hide the content until the user acts on it"

No need to imply a blur, clients will deal with this how they want

**Review by eskema — COMMENTED** [review]
only thing I'm unsure right now is NIP number, we must confirm with the oracle @fiatjaf

**Comment by fernandolguevara** [comment]
K, waiting oracle response :)

**Comment by fiatjaf** [comment]
@melvincarvalho asked to reserve this number for him, which we're doing for the time being.

Can you take NIP-36?

**Comment by melvincarvalho** [comment]
what about 617?

**Comment by fernandolguevara** [comment]
@fiatjaf @eskema 

NIP-36 :)

**Comment by fiatjaf** [comment]
Next time keep it as a draft please, we gotta give some time for our minds to collectively decide it is a final thing.

**Comment by fernandolguevara** [comment]
@fiatjaf as draft again

**Review by eskema — COMMENTED** [review]
looks good to me

**Comment by s3x-jay** [comment]
I've been working in the porn industry for roughly 15 years and I'm a "Community Ambassador" at XBiz.net (the leading B2B discussion forum for the adult industry). Based on that experience - I'm worried about NIP-36. I feel like it's incredibly important, but it can't achieve what it needs to achieve because there's no defined vocabulary for "reason". Not that "reason" should be completely locked down with a strict vocabulary, but 99% of the use cases are predictable and those cases would benefit greatly from a defined vocabulary that can be translated, etc.

Without going into too much detail there have been numerous attempts to implement classification systems over the years and all have failed. Many were culturally biased (e.g. "Unsuitable for persons under 18 years of age" is different in Sweden and Saudi Arabia). Other classification systems were just too complicated for the technology at the time. 

To me the classification system that had the most promise was ICRA, but it also failed. You can see it's vocabulary here…
https://web.archive.org/web/20080622002259/http://www.icra.org/vocabulary/

My suggestion would be to start with ICRA as a starting point and adjust it to fit the needs of Nostr.

For end users a Nostr client could just let them choose from the primary categories (which I've tweaked a bit):

- (Casual) Nudity
- Erotica ("soft core")
- Sex ("hard core")
- Violence
- Language
- Other potentially harmful topics

A Nostr client could show sub-categories once the user has chosen a main category - but that would be optional. (Keeping it simple is one of the lessons learned from the failure of ICRA.)

Automated systems would be encouraged to use a more detailed vocabulary (see below). The detailed vocabulary could go into a Nostr client's "preferences" section so users could specify whether certain types of content should 1) always be blocked, 2) never be blocked, or 3) the client should confirm with the user each time on a case-by-case basis. 

With the vocabulary below, I'm envisioning an automated system would have a comma delimited list of one or more of the items in the vocabulary. 

So for example, if I were posting "Someone in Chicago just faved this photo…" and I had an idea of what was in the photo I would use something like:

> Sex - Penetrative sex acts, Context - Porn - Gay male

If I were posting something from one of my sponsors and didn't know the exact nature of what was in the photo I might only use: 

> Context - Porn - Gay male

Or if I had a user who was uploading a picture I hadn't vetted but it was on a site which was typically sexually-explicit (e.g. a dating/hookup site or app) I might simply use:

> Context - Personal sexual expression/exploration

If I were posting something regarding a sexual health issue I might use:

> Nudity - Genitals/anus, Context - Medical


Here's my suggested vocabulary which builds on the concepts in the ICRA vocabulary…

-----

Context - Fine Art
Context - Educational
Context - Medical
Context - News
Context - Sports
Context - Religion
Context - Fantasy/fiction
Context - Fantasy/fiction - Video game

Context - Porn
Context - Porn - Heterosexual
Context - Porn - Gay male
Context - Porn - Lesbian
Context - Porn - Bisexual
Context - Porn - Transexual
Context - Porn - Transexual (feminine)
Context - Porn - Transexual (masculine)
Context - Porn - Gender fluid / non-binary

Context - Personal sexual expression/exploration


Nudity
Nudity - Breasts
Nudity - Buttocks
Nudity - Genitals/anus
Nudity - Other

Erotica
Erotica - Speaking/text only (no visuals)
Erotica - Physical Product
Erotica - Attire
Erotica - Kissing
Erotica - Erection (with no stimulation)
Erotica - Other

Sex
Sex - Speaking/text only (no visuals)
Sex - Obscured/implied sex acts
Sex - Masturbation
Sex - Non-penetrative sex acts
Sex - Penetrative sex acts
Sex - Hardcore fetish
Sex - Other


Violence
Violence - Assault/rape
Violence - Injury
Violence - Injury - human beings
Violence - Injury - animals
Violence - Injury - fantasy/animated characters
Violence - Blood and/or dismemberment
Violence - Blood and/or dismemberment - human beings
Violence - Blood and/or dismemberment - animals
Violence - Blood and/or dismemberment - fantasy/animated characters
Violence - Torture or killing
Violence - Torture or killing - human beings
Violence - Torture or killing - animals
Violence - Torture or killing - fantasy/animated characters
Violence - Other


Language
Language - Common expletives
Language - Profanity/swearing
Language - Abusive
Language - Other


Potentially harmful
Potentially harmful - Smoking 
Potentially harmful - Alcohol
Potentially harmful - Illegal drug use
Potentially harmful - Weapons
Potentially harmful - Gambling
Potentially harmful - Encourages life-threatening activities
Potentially harmful - Fear/intimidation/horror/terror
Potentially harmful - Encourages discrimination of protected minority
Potentially harmful - Other

-----

Choosing a main category would mean it could be any of the subcategories. Choosing an "Other" option means it's in that category, but not one of the specified subcategories. 

I don't mean the list above to be definitive. It should have input from the types of people who would use it and/or implement it. It's more of a starting place for discussion. It's quite important that it may be allowed to evolve over time.

Two other items that are  probably beyond the scope of NIP-36… 

1) It would be helpful if users could mark their profiles with the classification system above. Twitter has the rule of no sexual content in avatars and header pics so people don't come across offensive content accidentally. If people could classify their own profiles, then Nostr clients could blur the photos by default for those profiles (if the user has that set in their preferences). 

2) It might also be helpful if people could mark the content they see in their following/global feeds as sensitive. That classification could then be taken into consideration by others (in particular their followers).

**Comment by fernandolguevara** [comment]
@s3x-jay feel free to open an Issue/PR to express your concerns related to this nip...

I suggest it so speech does not remain in this hidden context (merged PR)

**Pull request merged**